source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -69,28 +52,27 @@ public PimsCompensationRequisition Update(PimsCompensationRequisition compensati
             _logger.LogInformation($"Updating Compensation Requisition with id ${compensationRequisition.CompensationRequisitionId}");
 
             var currentCompensation = _compensationRequisitionRepository.GetById(compensationRequisition.CompensationRequisitionId);
-            (bool? currentIsDraft, bool? newIsDraft) compReqStatusComparable = (currentIsDraft: currentCompensation.IsDraft, newIsDraft: compensationRequisition.IsDraft);
 
-            CheckDraftStatusUpdateAuthorized(compReqStatusComparable);
+            CheckDraftStatusUpdateAuthorized(currentCompensation.IsDraft, compensationRequisition.IsDraft);
             CheckTotalAllowableCompensation(compensationRequisition.AcquisitionFileId, compensationRequisition);
-            compensationRequisition.FinalizedDate = CheckFinalizedDate(compReqStatusComparable, currentCompensation.FinalizedDate);
+            compensationRequisition.FinalizedDate = CheckFinalizedDate(currentCompensation.IsDraft, compensationRequisition.IsDraft, currentCompensation.FinalizedDate);
 
             PimsCompensationRequisition updatedEntity = _compensationRequisitionRepository.Update(compensationRequisition);
-            AddNoteIfStatusChanged(compensationRequisition.Internal_Id, compensationRequisition.AcquisitionFileId, compReqStatusComparable);
+            AddNoteIfStatusChanged(compensationRequisition.Internal_Id, compensationRequisition.AcquisitionFileId, currentCompensation.IsDraft, compensationRequisition.IsDraft);
```

Comment:
I think this is where @eddherrera mentioned that using that tuple would capture the value of "IsDraft" 4 lines above here so we don't end up using updated values. Notice this check is done after `_compensationRequisitionRepository.Update(compensationRequisition);` so there's a chance of the IsDraft being changed by the call to Update and then the note would not be added. Perhaps @eddherrera can explain better.

If you really don't like the tuple then we would need to capture the value of IsDraft in separate values to preserve their original values before the Update - e.g.

```csharp
  bool = currentIsDraft = currentCompensation.IsDraft;
  bool = newIsDraft = compensationRequisition.IsDraft;
  ...
  _compensationRequisitionRepository.Update(compensationRequisition);

  // after update
  AddNoteIfStatusChanged(..., ..., currentIsDraft, newIsDraft);
  ...
```